src/decorators/index.ts

```diff
@@ -162,6 +162,22 @@ export function HasProgramWriteAccess(programIdExtractor: Function) {
   );
 }
 
+export function queryHasProgramWriteAccess(programId: string, egoToken: string) {
+  const decodedToken = verifyJwt(egoToken).data;
+
+  if (!decodedToken) {
+    throw new Error('This endpoint needs a valid authentication token');
+  }
+
+  const scopes = decodedToken.scopes;
+  const requiredScopes = [`PROGRAMDATA-${programId}.WRITE`, 'CLINICALSERVICE.WRITE'];
+  const hasWriteAccess = hasScope(requiredScopes, scopes);
+
+  if (!hasWriteAccess) throw new Error('Query does not have the required permissions');
+
+  return hasWriteAccess;
+}
+
```

Comment:
Would preferably update this file to use a HOC/decorator pattern
If `!hasWriteAccess` we throw an error, if `hasWriteAccess` you call the child function